llama-index-integrations/storage/chat_store/llama-index-storage-chat-store-upstash/llama_index/storage/chat_store/upstash/base.py

```diff
@@ -0,0 +1,113 @@
+from llama_index.core.llms import ChatMessage
+from llama_index.core.storage.chat_store.base import BaseChatStore
+
+import logging
+from typing import List, Optional, Any
+from upstash_redis import Redis
+from llama_index.core.bridge.pydantic import Field
+import json
+logger = logging.getLogger(__name__)
+
+# Convert a ChatMessage to a json object for Redis
+def _message_to_dict(message: ChatMessage) -> dict:
+    return message.dict()
+
+
+class UpstashChatStore(BaseChatStore):
+	"""Upstash chat store."""
+    
+	redis_client: Any = Field(description="Redis client.")
+	ttl: Optional[int] = Field(default=None, description="Time to live in seconds.")
+ 
+	def __init__(
+     	self,
+      	redis_url:str = "" ,
+       	redis_token:str = "", 
+        ttl: Optional[int]=None,
+    ):
+
+		if redis_url == "" or redis_token == "":
+			raise ValueError("Please provide a valid URL and token")
+
+		try:
+			self.redis_client = Redis(url=redis_url, token=redis_token)
+		except Exception:
+			logger.error("Upstash Redis client could not be initiated")
```

Comment:
we can catch as `except Exception as error`, and then include the error in the error we log. We could also raise the error in this. exception instead of simply logging an error.